src/models/Order.js

```diff
@@ -115,6 +117,30 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+// Middleware to send order completion email when the order status changes to "complete"
+OrderSchema.post('save', async function (doc) {
+  try {
+    if (doc.orderStatus === 'complete') {
+      const user = await User.findById(doc.user); // Assuming you have a User model
+
+      const orderItemsWithFullAlbum = await doc
+        .populate({ path: 'orderItems.album' })
+        .execPopulate();
+
+      // Send the order completion email
+      await sendOrderCompletedEmail(
+        user.email,
+        user.username, // Use the appropriate field for the username
```

Comment:
If we populate the order user with previous comment then these lines can instead be:

```suggestion
        orderItemsWithFullAlbum.user.email,
        orderItemsWithFullAlbum.user.username, // Use the appropriate field for the username
```